internal/abci/abci.go

```diff
@@ -825,8 +828,18 @@ func (a *AbciApp) FinalizeBlock(ctx context.Context, req *abciTypes.RequestFinal
 		}
 	}
 
-	// wait for all logs to be received
-	wg.Wait()
+	// wait for all logs to be received, or a premature shutdown
+	select {
+	case <-ctx.Done():
```

Comment:
Do we need to worry about this not being supported in comet v0.38? I don't think so because we close the notice subscribers when the database shuts down, but just want to double check that you see it the same

Reply:
No it's ok.  With 0.38 we only have one way to break out of here, and that's the notice signaling that the DB dies.  With 1.0 we can also get a signal from cometbft, but that's just going to be a bonus.